dashboard/test/controllers/teacher_dashboard_controller_test.rb

```diff
@@ -43,4 +43,22 @@ class TeacherDashboardControllerTest < ActionController::TestCase
     get :show, params: {section_id: cotaught_section.id}
     assert_response :success
   end
+
+  test 'redirect_to_newest_section: redirects to support URL if no sections instructed' do
+    other_teacher = create(:teacher)
+    sign_in other_teacher
+
+    get :redirect_to_newest_section
+
+    assert_redirected_to 'https://support.code.org/hc/en-us/articles/25195525766669-Getting-Started-New-Progress-View'
+  end
+
+  test 'redirect_to_newest_section: redirects to newest section progress page if sections instructed' do
+    sign_in @section_owner
+
+    get :redirect_to_newest_section
+
+    newest_section = @section_owner.sections_instructed.order(created_at: :desc).first
```

Comment:
Instead of getting the id using the same logic here, would it be possible to know which section was created last and call that section ID directly? Like it is, this doesn't actually test whether it retrieves the correct section, just whether it uses this same logic